lib/pbench/cli/agent/commands/results/move.py

```diff
@@ -0,0 +1,251 @@
+                        result_dir,
+                        temp_dir,
+                        self.context.controller,
+                        self.config,
+                        self.logger,
+                    )
+                except MakeResultTb.AlreadyCopied:
+                    self.logger.info(f"Already copied {result_dir}")
+                    continue
+                except MakeResultTb.BenchmarkRunning:
+                    self.logger.warning(
+                        f"Skipping {result_dir}: the benchmark appears to be"
+                        " running.  If that's incorrect, remove the"
+                        f" {result_dir}/.running directory and try again"
+                    )
+                    continue
+                except (NotADirectoryError, FileNotFoundError) as exc:
+                    self.logger.error(str(exc))
+                    failures += 1
+                    continue
+
+                try:
+                    result_tb_name, result_tb_len, result_tb_md5 = mrt.make_result_tb(
+                        single_threaded=single_threaded
+                    )
+                except BadMDLogFormat as exc:
+                    self.logger.warning(str(exc))
+                    failures += 1
+                    continue
+                except FileNotFoundError as exc:
+                    self.logger.error(str(exc))
+                    failures += 1
+                    continue
+                except RuntimeError as exc:
+                    self.logger.warning("Error encountered making tar ball, '%s'", exc)
+                    failures += 1
+                    continue
+                except Exception as exc:
+                    self.logger.error(
+                        "Unexpected error occurred making tar ball for '%s', '%s'",
+                        result_dir,
+                        exc,
+                    )
+                    failures += 1
+                    continue
+
+                try:
+                    crt = CopyResultTb(
+                        self.context.controller,
+                        result_tb_name,
+                        result_tb_len,
+                        result_tb_md5,
+                        self.config,
+                        self.logger,
+                    )
+                    crt.copy_result_tb(self.context.token)
+                except Exception as exc:
+                    if isinstance(exc, (CopyResultTb.FileUploadError, RuntimeError)):
+                        msg = "Error uploading file"
+                    else:
+                        msg = "Unexpected error occurred copying tar ball remotely"
+                    self.logger.error(f"{msg}, '%s', %s", result_tb_name, exc)
+                    failures += 1
+                    # We don't know why this operation failed; regardless,
+                    # trying to copy another tar ball remotely does not have
+                    # much chance of success.
```

Comment:
That's probably OK as a first pass. In reality, unless it's a connection failure or authorization failure from the server, that seems questionable. The bigger question is how often we'll get any other kind of error and whether it's worth the time and complication of decoding the possibilities.

Reply:
Okay, I can track this in an issue going forward.

Reply:
See issue #2429.